stdlib/string.mli

```diff
@@ -316,6 +316,17 @@ val equal: t -> t -> bool
 (** The equal function for strings.
     @since 4.03.0 *)
 
+val unseeded_hash : t -> int
```

Comment:
I think this function should be removed.

Reply:
Just to be clear I think we should provide `Hashtbl.no_seed` and people should use `M.hash Hashtbl.no_seed` with the seeded version to get this.